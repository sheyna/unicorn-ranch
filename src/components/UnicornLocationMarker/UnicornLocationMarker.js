import PropTypes from 'prop-types';

function UnicornLocationMarker(props) {
    const unic = props.unic;
    return (
        <li>{unic.name}
            <select
                name={unic.name + "-location"}
                id={unic.name}
                value={unic.location}
                onChange={(e) => props.moveUnicorn(e, unic.name)}
            >
                <option value="Barn">Barn</option>
                <option value="Pasture">Pasture</option>
                <option value="Trails">Trails</option>
            </select>
        </li>
    )
}

export default UnicornLocationMarker;

UnicornLocationMarker.propTypes = {
    unic: PropTypes.shape({
        name: PropTypes.string,
        location: PropTypes.string
    }),
    moveUnicorn: PropTypes.func
}