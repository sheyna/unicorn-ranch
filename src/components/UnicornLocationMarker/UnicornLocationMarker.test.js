import ReactTestRenderer from 'react-test-renderer';
import UnicornLocationMarker from "./UnicornLocationMarker";

const testUnicorn = {
    name: "Pink",
    location: "Barn"
}

const moveUnicorn = jest.fn();

describe('Home component', () => {
    it('should match snapshot', () => {
        const component = ReactTestRenderer.create(
            <UnicornLocationMarker
                unic={testUnicorn}
                moveUnicorn={moveUnicorn}
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});