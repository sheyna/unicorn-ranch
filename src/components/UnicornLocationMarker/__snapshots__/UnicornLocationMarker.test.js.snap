// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot 1`] = `
<li>
  Pink
  <select
    id="Pink"
    name="Pink-location"
    onChange={[Function]}
    value="Barn"
  >
    <option
      value="Barn"
    >
      Barn
    </option>
    <option
      value="Pasture"
    >
      Pasture
    </option>
    <option
      value="Trails"
    >
      Trails
    </option>
  </select>
</li>
`;
